### ISSUE TITLE
Doxyapp can't use "[source_file | source_dir]" parameter

### ISSUE THREAD

**Issue by hakua-doublemoon** [doxygen/doxygen#7109]
# Issue(as is)

* I used doxyapp as follows:

> ./doxyapp src/test.c

* Src directory had some c files. And, doxyapp made lists about all of src's files.
* This always happens.

# My Expects(to be)

* I want a list about only src/test.c.

# Environment

* Ubuntu (WIndows Subsystem for Linux)

# Resolution

* I think this issue is caused by the location of `checkConfiguration()` and `Config_getList(INPUT).append()`
    * Doxyapp don't has a Doxyfile. So after `checkConfiguration()`, all of Configs are initialized.
    * So I think I should rewrite it as below:

```
  checkConfiguration();
  Config_getList(INPUT).append(argv[1]); // insert this sentence!
  adjustConfiguration();
```

-----

I'm sorry my English is poor. Please let me know if you can't understand what I say or want more information.

**Comment by albert-github** [comment]
- Please specify the doxyapp version used.
- Which OS are you running on?
- did you make any changes ?
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent. (and also the commands given in the doxyapp).

The doxyapp is an example program, for what are you using it?

**Comment by hakua-doublemoon** [comment]
Hello,

I answer your question:

* Please specify the doxyapp version used.

> commit 34c9bac4624c5b94172af4a9d9168a44482fe552 (HEAD -> master, origin/master, origin/HEAD)
> Merge: 54f758e7 e40b8afa
> Author: Dimitri van Heesch <doxygen@gmail.com>
> Date:   Mon Jul 8 15:48:16 2019 +0200
> 
>     Merge pull request #7111 from albert-github/feature/issue_7110
> 
>     issue #7110 URL Encoding for Hyperlinks in PDF

* Which OS are you running on?

> :~/Windows/Documents/program/doxygen$ cat /etc/os-release
> NAME="Ubuntu"
> VERSION="18.04.2 LTS (Bionic Beaver)"
> ID=ubuntu
> ID_LIKE=debian
> PRETTY_NAME="Ubuntu 18.04.2 LTS"
> VERSION_ID="18.04"

* did you make any changes ?
    + no change

* Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent. (and also the commands given in the doxyapp).

    + bin.zip is. In bin.zip, log.txt is Operation log.

* The doxyapp is an example program, for what are you using it?

    + I want to verify doxygen comments. For this purpose, I research how to embed doxygen program in my program. And finally, I made preview program of doxygen:
    https://github.com/hakua-doublemoon/DoxygenPreviewer


[bin.zip](https://github.com/doxygen/doxygen/files/3375178/bin.zip)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7118

**Comment by hakua-doublemoon** [comment]
Thank you very much! It works perfect!

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
